test_conformance/run_conformance.py

```diff
@@ -63,19 +66,33 @@ def get_tests(filename, devices_to_test):
         comment = re.search("^#.*", line)
         if comment:
             continue
-        device_specific_match = re.search("^\s*(.+?)\s*,\s*(.+?)\s*,\s*(.+?)\s*$", line)
+        if sys.version_info[0] < 3:
+            device_specific_match = re.search("^\s*(.+?)\s*,\s*(.+?)\s*,\s*(.+?)\s*$", line)
+        else:
+            device_specific_match = re.search("^\s*(.+)\s*,\s*(.+)\s*,\s*(.+)\s*", line)
```

Comment:
Could you explain this change, please?  Is the "$" to match the end of line not important?  Why would it be needed with Python 2 but not with Python 3?

Why the change from the non-greedy `.+?` to the greedy `.+`?  I have tested the non-greedy `.+?` just now and it works fine with Python 3.6.9 and it works fine in my limited test case.  Why does it need changing here?  What sort of input text would "line" contain here, where this would make a difference?

Reply:
I'm not the developer that made this change and it's impossible to track who made the changes now. but seems that if I didn't change the regex, it would run into an error. See the following (the line numbers are not accurate as it's in a older version of this file.):

oem@payam-linux-nv32:~/nv32/conformance/2.0$ python3 run_conformance_kr.py svm.csv CL_DEVICE_TYPE_GPU
Testing on: ['CL_DEVICE_TYPE_GPU']
Traceback (most recent call last):
  File "run_conformance_kr.py", line 336, in <module>
    tests = get_tests(sys.argv[1], devices_to_test)
  File "run_conformance_kr.py", line 87, in get_tests
    test_path = str.replace(device_specific_match.group(3), '/', os.sep)
AttributeError: 'NoneType' object has no attribute 'group'
oem@payam-linux-nv32:~/nv32/conformance/2.0$

Reply:
Would you be able to share what the content of `line` was, here?

Reply:
[svm.csv](https://github.com/KhronosGroup/OpenCL-CTS/files/9609667/svm.csv)

This should be it, as line is read from the csv file.

Reply:
Might be the variable device_specific_match that causes the error in python 3 if we strip the whitespace. I double checked python documentation that every character in this search is exactly the same for python 2 and python 3. Will look into that.

Reply:
You are right - I didn't see why we need to change the regex search, thus I removed those lines. In addition I removed import string as we didn't use it. Please see my new commit. Thanks.

Reply:
Thanks for looking into this!